### DISCUSSION TITLE
[Feature Request] Add IP Geolocation to signature certificate

### DISCUSSION THREAD

**Discussion by Tom-H-L** [docusealco/docuseal#385] [Ideas]
At PandaDoc, on the signature certificate, besides having the IP of each signer displayed, also the geolocation of that IP is displayed. This is a nice feature that adds an additional layer of trust/ information/ plausibility check.

Of course, someone can pretend his geolocation by using a VPN. So it would be nice if there would be an additional probing if the IP is used by a well known VPN service and add a note to the geolocation, e.g.

IP Address: 185.220.101.4, Berlin, Germany (public VPN)

Here is a AI generated list of services that could be used for the geolocation feature:

1. IP2Proxy LITE - https://lite.ip2location.com/ip2proxy-lite
   Free monthly-updated database for open proxies.

2. IPinfo Free IP Database - https://ipinfo.io/products/free-ip-database
   Basic geolocation and ASN data, paid options for VPN detection.

3. X4BNet VPN Lists - https://github.com/X4BNet/lists_vpn
   GitHub lists of known VPN provider IP ranges.

4. GetIPIntel - https://getipintel.net/
   Free API to check if an IP is a VPN, proxy, or suspicious.

5. DB-IP Free - https://db-ip.com/
   Free IP geolocation database for general IP location data.

**Comment by nanoscopic** [comment]
@Tom-H-L 

How would you suggest integrating any of these things seamlessly into the system?

I've investigated geo-location a bit myself and generally it seems to require signing up and periodically downloading an updated DB to use. I'm not aware of any free system offhand that works at any sufficient scale as a free service.

Have you integrated with any of the things you have listed or ever actually used any of these things?

**Comment by Tom-H-L** [comment]
Closed because moved here: https://github.com/docusealco/docuseal/issues/409
